lib/network/net.go

```diff
@@ -185,15 +200,24 @@ func (c *TcpConn) Send(ctx context.Context, obj ProtocolMessage) error {
 	}
 	//c.Conn.SetWriteDeadline(time.Now().Add(timeOut))
 	// First write the size
-	packetSize := Size(len(b))
-	if err := binary.Write(c.Conn, globalOrder, packetSize); err != nil {
+	packetSize := uint64(len(b))
+
+	// FIXME Write the size as an UvarInt
+	var sb = make([]byte, binary.MaxVarintLen64)
+	n := binary.PutUvarint(sb[:], packetSize)
```

Comment:
That's not necessary, it does not change anything. if you look at the source of binary.Write ([code](https://golang.org/src/encoding/binary/binary.go?s=6397:6461#L224), it uses PutUVarint as the underlying function when you give it a uint64.